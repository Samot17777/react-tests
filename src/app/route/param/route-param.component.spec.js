import React from 'react';
import {render} from 'enzyme'
import {MemoryRouter, Route} from 'react-router-dom';
import {RouteParamComponent} from "./route-param.component";

describe('Route param component', () => {
    it('renders correctly', () => {
        const props = {
            match: {
                params: {
                    example: 'aa'
                }
            }
        };

        expect(
            render(
                <MemoryRouter initialEntries={['/foo']}>
                    <Route
                        path="/:example"
                        component={() => (
                            <RouteParamComponent  {...props}/>
                        )}
                    />
                </MemoryRouter>
            ),
        ).toMatchSnapshot();
    })
});