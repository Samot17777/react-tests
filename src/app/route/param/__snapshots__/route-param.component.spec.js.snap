// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route param component renders correctly 1`] = `
Array [
  <span>
    My route param is : aa 
  </span>,
  <span>
    My route param with router is : foo 
  </span>,
]
`;
